client/src/components/uploadImage.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import Dropzone from "react-dropzone";
+
+import {
+  Card,
+  CardActionArea,
+  CardMedia,
+  Tooltip,
+  Avatar,
+} from "@material-ui/core";
+
+const axios = require("axios");
+
+const useStyles = makeStyles((theme) => ({
+  card: {
+    width: "200px",
+    margin: "10px auto 0",
+    padding: "10px 0",
+  },
+  avatar: {
+    margin: "auto",
+    height: "170px",
+    width: "170px",
+    border: "5px solid rgba(0, 0, 0, 0.5)",
+    fontSize: "100px",
+  },
+}));
+
+const UploadImage = (props) => {
+  const classes = useStyles();
+
+  const [values, setValues] = useState({
+    id: "",
+    name: "",
+    imageUrl: "",
+  });
+
+  useEffect(() => {
+    axios.get("/account").then((res) => {
+      setValues({
+        id: res.data._id,
+        name: res.data.name,
+        imageUrl: res.data.imageUrl,
+      });
+    });
+  }, []);
+
+  const onDropAccepted = (acceptedFiles) => {
+    let data = new FormData();
+    data.append("id", values["id"]);
+    data.append("image", acceptedFiles[0]);
+
+    axios
+      .post("/uploadImage", data)
+      .then((res) => {
+        // TODO: Reload the image without having to refresh the entire page
+        setValues({ id: values["id"], imageUrl: res.data.imageUrl });
+        window.location.reload();
```

Comment:
It looks like when you call <UploadImage /> You're passing in `values` as a prop. Also pass in `setValues`, then remove the `useState` in this component. Now replace these lines with the `setValues` from the prop. That should update the component correctly. (Hopefully)

Reply:
I switched so the UploadImage component was working off of props passed from the Profile page, but that did not end up fixing the issue. What ultimately solved it was appending Date.now() to the image src to force refresh (since the src technically doesn't change).

Reply:
I was just wrestling the same issue, what I had settled on was creating adding an invisible element to the page and updating that:
```
const [updateView, setUpdateView] = useState(0);

return (
  <p style={{ display: "none" }}>{updateView}</p>
);
```
then you can just call `setUpdateView(updateView + 1)` whenever you want to update